spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewritePositionDeleteFilesSparkAction.java

```diff
@@ -458,6 +460,7 @@ public int totalGroupCount() {
   }
 
   private StructLike coercePartition(PositionDeletesScanTask task, StructType partitionType) {
-    return PartitionUtil.coercePartition(partitionType, task.spec(), task.partition());
+    Types.StructType dedupType = PositionDeletesTable.partitionType(table.schema(), partitionType);
```

Comment:
This was using the original partition type, but to coerce partition here (for grouping), we actually need the final partition type (with reassigned field ids).